src/main/java/io/beanmapper/config/CoreConfiguration.java

```diff
@@ -470,4 +472,38 @@ public BeanConverterStore getBeanConverterStore() {
     public <S, T> BeanConverter getBeanConverter(Class<S> source, Class<T> target) {
         return this.beanConverterStore.get(source, target);
     }
+
+    @Override
+    public boolean isInDiagnosticsMode() {
+        return false;
+    }
+
+    @Override
+    public int getMappingDepth() {
+        return 0;
+    }
+
+    @Override
+    public Configuration getParentConfiguration() {
+        throw new BeanMapperDiagnosticsDisabledException("CoreConfiguration does not support parent configuration.");
+    }
+
+    @Override
+    public DiagnosticsNode getBeanMapperDiagnostics() {
+        return null;
+    }
+
+    @Override
+    public void setBeanMapperDiagnostics(DiagnosticsNode diagnostics) {
+        throw new BeanMapperDiagnosticsDisabledException("CoreConfiguration does not support bean mapper diagnostics.");
+    }
+
+    public int getDepth() {
+        return 0;
```

Comment:
icm met de Optional returning methods in de interface, kan dit naar de default implementatie.